### PR TITLE
Working LinkedIn OAuth response

### PR DESCRIPTION
**Pull request by ivanfoong**
Working linkedin oauth redirect and fetching `lite_profile` and `email` with the returned token

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -48,23 +40,23 @@ func main() {
 		}
 	})
 
-  /*
-	auth0Domain := os.Getenv("AUTH0_DOMAIN")
-	auth0ClientID := os.Getenv("AUTH0_CLIENT_ID")
-	auth0Audience := os.Getenv("AUTH0_AUDIENCE")
-	auth0Callback := os.Getenv("AUTH0_CALLBACK")
+	/*
+		auth0Domain := os.Getenv("AUTH0_DOMAIN")
```
Comment (subhransu) [inline main.go]: let's remove this code?

File: api/controllers/oauth-controller.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	err = json.Unmarshal(str, &profile)
+	if err != nil {
+		return LiteProfile{}, err
+	}
+
+	return profile, nil
+}
+
+func getEmail(client *http.Client, accessToken string) (LinkedInEmailResponse, error) {
+	req, err := http.NewRequest("GET", "https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))", nil)
```
Comment (subhransu) [inline api/controllers/oauth-controller.go:106]: can we get the URL from a config file?

File: api/controllers/oauth-controller.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		ClientID:     os.Getenv("LINKEDIN_CLIENT_ID"),
+		ClientSecret: os.Getenv("LINKEDIN_CLIENT_SECRET"),
+		RedirectURL:  fmt.Sprintf("%s/oauth/linkedin_callback", os.Getenv("LINKEDIN_CALLBACK_HOST")),
+		Scopes:       []string{"r_emailaddress", "r_liteprofile"},
+		Endpoint:     linkedin.Endpoint,
+	}
+}
+
+func getProfile(client *http.Client, accessToken string) (LiteProfile, error) {
+	req, err := http.NewRequest("GET", "https://api.linkedin.com/v2/me?projection=(id,firstName,lastName,email,profilePicture(displayImage~:playableStreams))", nil)
```
Comment (subhransu) [inline api/controllers/oauth-controller.go:79]: same here ... we can get the BASE_URL and the scopes or URL path separately.

**Review by subhransu — CHANGES_REQUESTED** [review]
few minor comments. rebase with the master as well.

File: api/controllers/oauth-controller.go (hunk 2 above)
Comment (ivanfoong) [reply api/controllers/oauth-controller.go:106]: Let's leave it to @kajalsinha to decide what's the best way to implement config file

File: api/controllers/oauth-controller.go (hunk 3 above)
Comment (ivanfoong) [reply api/controllers/oauth-controller.go:79]: ditto

**Review by subhransu — APPROVED** [review]

**Comment by subhransu** [comment]
Ignore the build failure. If it's from a forked repo it will not fetch the secrets from GitHub vault. 
Probably next time just create a branch on this repo?

**Pull request merged**